src/chat.rs

```diff
@@ -74,37 +208,38 @@ impl Chat {
                 Err(err.into())
             }
             Ok(mut chat) => {
-                match chat.id {
-                    DC_CHAT_ID_DEADDROP => {
-                        chat.name = context.stock_str(StockMessage::DeadDrop).into();
-                    }
-                    DC_CHAT_ID_ARCHIVED_LINK => {
-                        let tempname = context.stock_str(StockMessage::ArchivedChats);
-                        let cnt = dc_get_archived_cnt(context);
-                        chat.name = format!("{} ({})", tempname, cnt);
-                    }
-                    DC_CHAT_ID_STARRED => {
-                        chat.name = context.stock_str(StockMessage::StarredMsgs).into();
-                    }
-                    _ => {
-                        if chat.typ == Chattype::Single {
-                            let contacts = get_chat_contacts(context, chat.id);
-                            let mut chat_name = "Err [Name not found]".to_owned();
-
-                            if let Some(contact_id) = contacts.first() {
-                                if let Ok(contact) = Contact::get_by_id(context, *contact_id) {
-                                    chat_name = contact.get_display_name().to_owned();
-                                }
-                            }
+                if chat.id.is_deaddrop() {
+                    chat.name = context.stock_str(StockMessage::DeadDrop).into();
+                } else if chat.id.is_archived_link() {
+                    let tempname = context.stock_str(StockMessage::ArchivedChats);
+                    let cnt = dc_get_archived_cnt(context);
+                    chat.name = format!("{} ({})", tempname, cnt);
+                } else if chat.id.is_starred() {
+                    chat.name = context.stock_str(StockMessage::StarredMsgs).into();
+                } else {
+                    if chat.typ == Chattype::Single {
+                        let contacts = get_chat_contacts(context, chat.id);
+                        let mut chat_name = "Err [Name not found]".to_owned();
 
-                            chat.name = chat_name;
+                        if let Some(contact_id) = contacts.first() {
+                            if let Ok(contact) = Contact::get_by_id(context, *contact_id) {
+                                chat_name = contact.get_display_name().to_owned();
+                            }
                         }
 
-                        if chat.param.exists(Param::Selftalk) {
-                            chat.name = context.stock_str(StockMessage::SavedMessages).into();
-                        } else if chat.param.exists(Param::Devicetalk) {
-                            chat.name = context.stock_str(StockMessage::DeviceMessages).into();
+                        if !(*contacts).is_empty() {
```

Comment:
This seems like a merge conflict! Same as the code above, but written in a less rusty way.

Reply:
wow, yes.  strange how that go there.  scary how it compiled and passed all tests